packages/jira-adapter/src/change_validators/field_contexts/second_global_context.ts

```diff
@@ -71,9 +71,43 @@ export const fieldSecondGlobalContextValidator: ChangeValidator = async (changes
     .filter(instance => instance.value.isGlobalContext)
     .toArray()
 
+  await fillFieldToGlobalContextCount()
+  await awu(changes).filter(isInstanceChange)
+    .filter(change => getChangeData(change).elemID.typeName === PROJECT_TYPE)
+    .filter(isRemovalOrModificationChange)
+    .forEach(change => {
+      change.data.before.value.fieldContexts
+        .filter(isReferenceExpression)
+        .forEach(async (context: ReferenceExpression) => {
+          const fieldElemId = await getParentElemID(await context.getResolvedValue())
+          fieldContextToChange.set(context.elemID.getFullName(), { change, fieldElemId })
+        })
+    })
+  const projectInstances = await getInstancesFromElementSource(elementSource, [PROJECT_TYPE])
+
+  // if there is a project that using a field context that is not global context
+  projectInstances.flatMap(instance => instance.value.fieldContexts)
+    .filter(isReferenceExpression)
+    .forEach((context: ReferenceExpression) => {
+      fieldContextToChange.delete(context.elemID.getFullName())
+    })
+  const errorMessages = Array.from(fieldContextToChange.keys())
+    .filter(context => fieldContextToChange.get(context) !== undefined)
+    .filter(context => {
+      const changeAndFieldElemId = fieldContextToChange.get(context)
+      if (changeAndFieldElemId !== undefined) {
+        const { fieldElemId } = changeAndFieldElemId
+        // if there is another global context for the field
+        return fieldToGlobalContextCount[fieldElemId.getFullName()] === 1
+      }
+      return false
```

Comment:
styling nit: early return (+ log) for unexpected cases
```suggestion
      if (changeAndFieldElemId === undefined) {
        log.error(...)
        return false
      }
      const { fieldElemId } = changeAndFieldElemId
      // if there is another global context for the field
      return fieldToGlobalContextCount[fieldElemId.getFullName()] === 1
      }
```